### DISCUSSION TITLE
Running graphics cards on raspberry pi through windows 10

### DISCUSSION THREAD

**Discussion by kamleshboi69** [geerlingguy/raspberry-pi-pcie-devices#171] [Ideas] [answered]
Hi guys so I thought how about running graphics cards through windows 10 on raspberry pi 4 I think it would give better information about fails and easy troubleshooting  You guys can download windows through Wor website (windows on raspberry) that's it thank you bye.

**Comment by geerlingguy** [comment]
Many people have suggested running Windows to try to get graphics cards working, but the problem there is that almost nobody running Windows on ARM has done any work to support ARM drivers for GPUs (maybe nobody ever?), and Microsoft doesn't build ARM drivers for Nvidia, AMD, etc. (at least that I'm aware of) yet... so I don't think it would be a very fruitful endeavor at this time.

**Comment by Coreforge** [comment, marked as answer]
Using windows instead of linux isn't going to help in this case as the issue is more of a hardware one than software. The hardware doesn't support some features the driver normally expects, so the driver has to be changed to work around those limitations. Troubleshooting would also be nearly impossible on windows as the drivers aren't open source AFAIK, so it's impossible to poke around in it and dump data at various places or make any changes at all.

**Comment by brokensyntax** [comment]
Recently Rene Rebe (RISC-V) has done some Kernel patching and work to get at least partial functionality from modern Graphics cards in ARM platforms.

Anyone looking at this work to get compatibility/functionality on RPi?
https://www.youtube.com/watch?v=Sv4-_a_3BKg

**Reply by geerlingguy** [reply]
This has been discussed over in https://github.com/geerlingguy/raspberry-pi-pcie-devices/issues/6#issuecomment-885541054
